integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonCreateTableAsSelectCommand.scala

```diff
@@ -54,8 +54,8 @@ case class CarbonCreateTableAsSelectCommand(
     setAuditTable(dbName, tableName)
     setAuditInfo(Map("query" -> query.simpleString))
     // check if table already exists
-    if (sparkSession.sessionState.catalog.listTables(dbName)
-      .exists(_.table.equalsIgnoreCase(tableName))) {
+    if (!sparkSession.sessionState.catalog
+      .tableExists(TableIdentifier(tableName, Some(dbName)))) {
```

Comment:
We need to check if the table exists, currently because of the ! check the condition is reversed.

Reply:
I'm sorry. It's my mistake.

What I mean is It's very bad to use `listTables` to check table's existence instead of `tableExists`.

I've rebased and fixed this.